Unity-Technologies/com.unity.multiplayer.samples.coop
# fix: not returning to mainmenu when reconnect fails

### Description (*)

This PR does three things: first it adds null checks to prevent null reference exceptions, then it adds a mechanism for the host to tell the clients when it voluntarily quits a game, so that if they quit using the in-game quit button, they will not attempt to reconnect and simply display a message instead. Finally, it reduces the number of reconnection attempts to 1 to avoid an issue that seems linked to Relay, until further investigation allows us to better understand that bug (see [MTT-2794](https://jira.unity3d.com/browse/MTT-2794) for details)
### Related Pull Requests

### Issue Number(s) (*)

Fixes issue(s): [MTT-2786](https://jira.unity3d.com/browse/MTT-2786) and [MTT-2795](https://jira.unity3d.com/browse/MTT-2795)
### Manual testing scenarios

1. Start a game as a host
2. Join with a client
3. With the host, leave the session via the in-game quit button
4. See the client is disconnected and returned to the main menu and doesn't try to reconnect
5. Start another game with a host and client
6. End the play mode if in editor or quit the application if using a build (not via the in-game quit button)
7. See the client attempting to reconnect, then failing and going back to the main menu
### Questions or comments

On hold because will have to wait until #523 is merged first, as this is depending on a fix introduced by this other PR.
### Contribution checklist
